internal/notification/types/user_phone.go

```diff
@@ -7,35 +7,61 @@ import (
 
 	"github.com/zitadel/zitadel/internal/eventstore"
 	"github.com/zitadel/zitadel/internal/notification/messages"
+	"github.com/zitadel/zitadel/internal/notification/templates"
 	"github.com/zitadel/zitadel/internal/query"
 	"github.com/zitadel/zitadel/internal/zerrors"
 )
 
+type serializableData struct {
+	templates.TemplateData
+
+	Args map[string]interface{}
+}
+
```

Comment:
There needs to be at least some more metadata (which event led to this / what's the intent of the notification). It's otherwise hard to handle it properly

additionally, formatting (case) might be a topic to discuss as well

currently i get for phone verification:
```json
{
   "TemplateData":{
      "Title":"Zitadel - Verify phone",
      "PreHeader":"Verify phone",
      "Subject":"Verify phone",
      "Greeting":"Hello ZITADEL Admin,",
      "Text":"A new phone number has been added. Please use the following code to verify it W50XNG.",
      "URL":"",
      "ButtonText":"Verify phone",
      "PrimaryColor":"#5469d4",
      "BackgroundColor":"#fafafa",
      "FontColor":"#000000",
      "LogoURL":"",
      "FontURL":"",
      "FontFaceFamily":"",
      "FontFamily":"-apple-system, BlinkMacSystemFont, Segoe UI, Lato, Arial, Helvetica, sans-serif",
      "IncludeFooter":false,
      "FooterText":"VerifyPhone.Footer"
   },
   "Args":{
      "ChangeDate":"2024-09-06T08:47:49.146249+02:00",
      "Code":"W50XNG",
      "CreationDate":"2024-07-30T15:48:23.755112+02:00",
      "DisplayName":"ZITADEL Admin",
      "Domain":"localhost",
      "FirstName":"ZITADEL",
      "LastEmail":"livio@zitadel.com",
      "LastName":"Admin",
      "LastPhone":"+41791234567",
      "LoginNames":[
         "zitadel-admin@zitadel.localhost"
      ],
      "NickName":"",
      "PreferredLoginName":"zitadel-admin@zitadel.localhost",
      "UserName":"zitadel-admin@zitadel.localhost",
      "VerifiedEmail":"livio@zitadel.com",
      "VerifiedPhone":""
   }
}
```

and OTP SMS:
```json
{
   "TemplateData":{
      "Title":"VerifySMSOTP.Title",
      "PreHeader":"VerifySMSOTP.PreHeader",
      "Subject":"VerifySMSOTP.Subject",
      "Greeting":"VerifySMSOTP.Greeting",
      "Text":"14533756 is your one-time-password for localhost. Use it within the next 5m0s.\n@localhost #14533756",
      "URL":"",
      "ButtonText":"VerifySMSOTP.ButtonText",
      "PrimaryColor":"#5469d4",
      "BackgroundColor":"#fafafa",
      "FontColor":"#000000",
      "LogoURL":"",
      "FontURL":"",
      "FontFaceFamily":"",
      "FontFamily":"-apple-system, BlinkMacSystemFont, Segoe UI, Lato, Arial, Helvetica, sans-serif",
      "IncludeFooter":false,
      "FooterText":"VerifySMSOTP.Footer"
   },
   "Args":{
      "ChangeDate":"2024-09-06T08:49:46.079596+02:00",
      "CreationDate":"2024-07-30T15:48:23.755112+02:00",
      "DisplayName":"ZITADEL Admin",
      "Domain":"localhost",
      "Expiry":300000000000,
      "FirstName":"ZITADEL",
      "LastEmail":"livio@zitadel.com",
      "LastName":"Admin",
      "LastPhone":"+41791234567",
      "LoginNames":[
         "zitadel-admin@zitadel.localhost"
      ],
      "NickName":"",
      "OTP":"14533756",
      "Origin":"http://localhost:8080",
      "PreferredLoginName":"zitadel-admin@zitadel.localhost",
      "UserName":"zitadel-admin@zitadel.localhost",
      "VerifiedEmail":"livio@zitadel.com",
      "VerifiedPhone":"+41791234567"
   }
}
```